src/highrl/agents/agent.py

```diff
@@ -193,55 +168,56 @@ def compute_position(self, action: Any, delta_t: float) -> Tuple:
+        x_pos = self.pos.x + velocity * np.cos(self.theta + angle) * delta_t
+        y_pos = self.pos.y + velocity * np.sin(self.theta + angle) * delta_t
         theta = self.fix(self.theta + action.w * delta_t, 2 * np.pi)
 
-        return px, py, theta
+        return x_pos, y_pos, theta
 
-    def step(self, action: ActionXY, delta_t: float) -> None:
-        """Performs an action and update the agent state.
-
-        Args:
-            action (List): action decided by the agent model but in ActionXY object format
-            delta_t (float): time difference between actions
-        """
-        self.check_validity(action)
-        pos = self.compute_position(action, delta_t)
-        self.px, self.py, self.theta = pos
-        self.vx = action.vx
-        self.vy = action.vy
-        self.w = action.w
-
-    def fix(self, x, mod):
+    def fix(self, base: Union[int, float], mod: Union[int, float]) -> Union[int, float]:
         """Fix input x to be in range [0:mod-1].
 
         Args:
-            x (int | float): input range
-            mod (int | float): modulus
+            base (Union[int, float]): input range
+            mod (int): modulus
 
         Returns:
-            int | float: input with desired range
+            Union[int, float]: Input with desired range
         """
-        while x < 0:
-            x += mod
-        while x >= mod:
-            x -= mod
-        return x
+        while base < 0:
+            base += mod
+        while base >= mod:
+            base -= mod
+        return base
 
     def reached_destination(self) -> bool:
         """Determines if agent reached the goal postion.
 
         Returns:
             bool: whether the agent has reached the goal or not
         """
-        robot_pos = np.array(self.get_position())
-        goal_pos = np.array(self.get_goal_position())
-        agent_goal_dist = robot_pos - goal_pos
         min_allowed_dist = self.radius + self.goal_radius
-        return norm(agent_goal_dist).item() < min_allowed_dist
+        return self.dist_to_goal() < min_allowed_dist
+
+    def dist_to_goal(self) -> float:
+        """Compute the distance from the agent to the goal"""
+        return norm(self.pos.get_coords() - self.gpos.get_coords()).item()
+
+    def step(self, action: ActionXY, delta_t: float) -> None:
+        """Performs an action and update the agent state.
+        Args:
+            action (List): action decided by the agent model but in ActionXY object format
+            delta_t (float): time difference between actions
+        """
+        self.check_validity(action)
+        pos = self.compute_position(action, delta_t)
+        x_pos, y_pos, self.theta = pos
+        self.pos.set_pos(x_pos, y_pos)
+        self.vx = action.vx
+        self.vy = action.vy
+        self.w = action.w
 
     def is_overlapped(self, obstacle: SingleObstacle, check_target: str = "agent"):
         """Checks if overlap between the agent/goal and an obstacle.
```

Comment:
Replace doc-string with `"""Checks if there is overlap between the agent/goal and a given obstacle"""`